src/DIPS.Xamarin.UI/Controls/Toast/Toast.cs

```diff
@@ -0,0 +1,274 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using DIPS.Xamarin.UI.Internal.Xaml;
+using Xamarin.Essentials;
+using Xamarin.Forms;
+
+namespace DIPS.Xamarin.UI.Controls.Toast
+{
+    /// <summary>
+    ///     Toast control that would appear on top of the presented view
+    /// </summary>
+    public class Toast
+    {
+        private Toast()
+        {
+            Application.Current.PageAppearing -= OnPageAppearing;
+            Application.Current.PageAppearing += OnPageAppearing;
+
+            Application.Current.PageDisappearing -= OnPageDisappearing;
+            Application.Current.PageDisappearing += OnPageDisappearing;
+        }
+
+        /// <summary>
+        ///     Get the current instance of the Toast control
+        /// </summary>
+        public static Toast Current { get; } = new Toast();
+
+        private CancellationTokenSource CancellationSource { get; set; } = new CancellationTokenSource();
+        private Dictionary<string, Grid> ToastContainers { get; } = new Dictionary<string, Grid>();
```

Comment:
This will cache UI elements forever as Current is a static instance and no remove is done. 
Remove items fro this to prevent memory leaks. I recommend testing with code to generate pages and poping/pushing while looking at the memory consumption of the app, I suspect it would increase as no pages can dispose.

Reply:
Yes, should remove from this too

Reply:
Done